### ISSUE TITLE
Enhancement - Make lottery loot table based

### ISSUE THREAD

**Issue by psiber-on-uncovery** [uncovery/uncovery_me#148]
<theotherrealpsib> use 100 uncs as the base item, then everything else is chance x in y

<theotherrealpsib> and it jsut iterates through the registered items
<theotherrealpsib> you might get nothing, might get 2 or 3 things
<theotherrealpsib> you jsut tweak the drop rates until results as desired
<uncovery> sounds good

<theotherrealpsib> it allows for some cooler things to be done too, like if you had a function passed as an optional argument to the reward, you could make a reward run a function.

<theotherrealpsib> ill add it as an open issue to convert it to loot tables. that way it will remind me.
<theotherrealpsib> "enhancement"

**Comment by uncovery** [comment]
this should include:
 - a check if someone is entitled to get a specific reward. If not, the chance to win something should not be reduced. The issue here is that it would be complicated to first go through all the rewards and check if the user can win it, and then assemble the list. Instead, after a reward is chosen, the system needs to check if the user can get it and if not, take something else. Theoretically, the system could restart until a good reward can be found. Like this we would give homes only to those who have available slots or give special rewards to donators etc.
- be modular so plugins can ad rewards dynamically
- a more dynamic way to get blocks, ores and manufactured blocks. Maybe we can make the chances to win those in line with the difficulty to make them. in the extreme, we get a mining-probability-list for ores and raw materials and multiply them with the amount of force it takes to mine a block, and the combined stuff it needs to make a manufactured block. Not sure if we can get that data somewhere. maybe from minecraft itself?

**Comment by psiber-on-uncovery** [comment]
homes are already checked to ensure they are below maximum home count. umc_home_add runs a maximum check.

**Comment by uncovery** [comment]
Yes but it's too late since we cannot go back and pick something else if it fails.

**Comment by psiber-on-uncovery** [comment]
ahhhh, right. I get you. Yes this makes sense then, like a minimum reward amount.

I like it.
